homework_3/events/index.js

```diff
@@ -0,0 +1,97 @@
+const stream = require('stream');
+const fs = require('fs');
+const path = require('path');
+const csv = require('csv-parser');
+
+// --- --- --- --- --- --- --- Handlers --- --- --- --- --- --- --- ---
+
+getAll = async(req, res) => readEvents().then(events => {
+    const streamEvents = stream.Readable.from(JSON.stringify(events));
+    streamEvents.pipe(res);
+});
+
+getEvents = async(req, res) => {
+    const location = req.query.location;
+    readEvents().then(events => {
+        if (location)
+            events = events.filter(event => event.location === location);
+        res.json(events);
+    });
```

Comment:
We might add `catch` statement. In case of smth going wrong, we would still like to send some `error-response` to a client.